src/plugins/data_view_field_editor/public/service/field_format_editors/field_format_editors.ts

```diff
@@ -8,24 +8,31 @@
 
 import { FieldFormatEditorFactory } from '../../components/field_format_editor';
 
+export interface FieldFormatEditorStart {
+  getAll: () => FieldFormatEditorFactory[];
+  getById: <P>(id: string) => FieldFormatEditorFactory<P> | undefined;
+}
+
 export class FieldFormatEditors {
   private editors: FieldFormatEditorFactory[] = [];
 
   public setup(defaultFieldEditors: FieldFormatEditorFactory[] = []) {
     this.editors = defaultFieldEditors;
 
     return {
-      register: (editor: FieldFormatEditorFactory) => {
-        this.editors.push(editor);
+      register: <P>(editor: FieldFormatEditorFactory<P>) => {
+        this.editors.push(editor as FieldFormatEditorFactory<unknown>);
```

Comment:
Its not clear to me what these additional types are doing. Is it okay to go without them?

Reply:
I battled around with this a  bit, and I'd love if someone could double-check me on this. The generics allow
calling code to pass in a typed FieldFormatEditorFactory -- otherwise it will only allow `FieldFormatEditorFactory<unknown>` as an argument to `.register()`.

Reply:
This is something I made progress on, likely producing an undesirable line of code or two. Definitely feel like I'm ready for someone who's seen this before to show me how its done. It seems pretty simple - we want an array of more specific types to be seen as the more general type. Maybe that has to be done with 'as'.